pkg/receive/hashring.go

```diff
@@ -108,7 +119,53 @@ func NewHashring(matcher Matcher, groups []*targetgroup.Group) Hashring {
 		matcher:   matcher,
```

Comment:
I thought about this again, and I feel like this should be pushed further down into the hashring itself, which will allow it to be more extensible in the future.